recipe/meta.yaml

```
@@ -1,54 +1,76 @@
-{% set version = "3.141.0" %}
+{% set version = "4.24.0" %}
 
 package:
   name: selenium
   version: {{ version }}
-
+#  Added 0000-no-rust-setup.patch to remove all references to Rust
+#  extensions from setup.py. This includes removing the import of setuptools_rust and
+#  the rust_extensions configuration, which resolves build failures caused by the unnecessary
+#   inclusion of Rust extensions.
 source:
+  patches:
+    - 0000-no-rust-setup.patch
   url: https://pypi.org/packages/source/s/selenium/selenium-{{ version }}.tar.gz
-  sha256: deaf32b60ad91a4611b98d8002757f29e6f2c2d5fcaf202e1c9ad06d6772300d
+  sha256: 88281e5b5b90fe231868905d5ea745b9ee5e30db280b33498cc73fb0fa06d571
 
 build:
-  number: 1000
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed -vvv
+  skip: True  # [py<38]
   missing_dso_whitelist:  # [osx]
     - '*'                 # [osx]
 
 requirements:
   build:
     - {{ compiler('c') }}
+    - patch       # [not win]
+    - m2-patch    # [win]
   host:
     - python
     - pip
+    - setuptools
+    - wheel
   run:
     - python
-    - urllib3
+    - urllib3 >=1.26,<3
+    - trio >=0.17,<1.0
+    - trio-websocket >=0.9,<1.0
+    - certifi >=2021.10.8
+    - typing_extensions >=4.9,<5.0
+    - websocket-client >=1.8,<2.0
 
 test:
+  requires:
+    - pip
+  commands:
+    - pip check
   imports:
     - selenium
     - selenium.common
-    - selenium.webdriver
-    - selenium.webdriver.android
-    - selenium.webdriver.chrome
     - selenium.webdriver.common
-    - selenium.webdriver.common.html5
+    - selenium.webdriver.chrome
+    - selenium.webdriver.chromium
     - selenium.webdriver.edge
     - selenium.webdriver.firefox
-    - selenium.webdriver.ie
-    - selenium.webdriver.opera
-    - selenium.webdriver.phantomjs
     - selenium.webdriver.remote
+    - selenium.webdriver.ie
+    - selenium.webdriver.safari
     - selenium.webdriver.support
+    - selenium.webdriver.webkitgtk
+    - selenium.webdriver.wpewebkit
 
 about:
-  home: http://www.seleniumhq.org/
+  home: https://www.seleniumhq.org/
   license_file: {{ RECIPE_DIR }}/LICENSE
-  license: Apache 2.0
+  license: Apache-2.0
   license_family: Apache
   summary: 'Python bindings for the Selenium WebDriver for automating web browser interaction.'
+  description: |
+    Selenium specifically provides an infrastructure for the W3C WebDriver
+    specification — a platform and language-neutral coding interface compatible
+    with all major web browsers.
   dev_url: https://github.com/SeleniumHQ/selenium/
-  doc_url: http://www.seleniumhq.org/docs/
+  doc_url: https://www.seleniumhq.org/docs/
```

```suggestion
  doc_url: https://www.selenium.dev/documentation/
```
Because of a redirect